pkg/sdk/testint/tags_integration_test.go

```diff
@@ -281,3 +283,915 @@ func TestInt_TagsShowByID(t *testing.T) {
+			require.NoError(t, err)
+
+			_, err = client.SystemFunctions.GetTag(ctx, tag.ID(), id, tc.objectType)
+			require.ErrorContains(t, err, "sql: Scan error on column index 0, name \"TAG\": converting NULL to string is unsupported")
+		})
+	}
+}
+
+func createDatabaseFromShare(t *testing.T) (*sdk.Database, func()) {
+	t.Helper()
+	client := testClient(t)
+	secondaryClient := testSecondaryClient(t)
+	ctx := testContext(t)
+
+	shareTest, shareCleanup := secondaryTestClientHelper().Share.CreateShare(t)
+	t.Cleanup(shareCleanup)
+
+	sharedDatabase, sharedDatabaseCleanup := secondaryTestClientHelper().Database.CreateDatabase(t)
+	t.Cleanup(sharedDatabaseCleanup)
+
+	databaseId := sharedDatabase.ID()
+
+	err := secondaryClient.Grants.GrantPrivilegeToShare(ctx, []sdk.ObjectPrivilege{sdk.ObjectPrivilegeUsage}, &sdk.ShareGrantOn{
+		Database: sharedDatabase.ID(),
+	}, shareTest.ID())
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := secondaryClient.Grants.RevokePrivilegeFromShare(ctx, []sdk.ObjectPrivilege{sdk.ObjectPrivilegeUsage}, &sdk.ShareGrantOn{
+			Database: sharedDatabase.ID(),
+		}, shareTest.ID())
+		require.NoError(t, err)
+	})
+
+	err = secondaryClient.Shares.Alter(ctx, shareTest.ID(), &sdk.AlterShareOptions{
+		IfExists: sdk.Bool(true),
+		Set: &sdk.ShareSet{
+			Accounts: []sdk.AccountIdentifier{
+				testClientHelper().Account.GetAccountIdentifier(t),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	err = client.Databases.CreateShared(ctx, databaseId, shareTest.ExternalID(), &sdk.CreateSharedDatabaseOptions{})
+	require.NoError(t, err)
+
+	database, err := client.Databases.ShowByID(ctx, databaseId)
+	require.NoError(t, err)
+
+	return database, testClientHelper().Database.DropDatabaseFunc(t, database.ID())
+}
+
+func createDatabaseReplica(t *testing.T) (*sdk.Database, func()) {
+	t.Helper()
+	client := testClient(t)
+	secondaryClient := testSecondaryClient(t)
+	ctx := testContext(t)
+
+	sharedDatabase, sharedDatabaseCleanup := secondaryTestClientHelper().Database.CreateDatabase(t)
+	t.Cleanup(sharedDatabaseCleanup)
+
+	err := secondaryClient.Databases.AlterReplication(ctx, sharedDatabase.ID(), &sdk.AlterDatabaseReplicationOptions{
+		EnableReplication: &sdk.EnableReplication{
+			ToAccounts: []sdk.AccountIdentifier{
+				testClientHelper().Account.GetAccountIdentifier(t),
+			},
+			IgnoreEditionCheck: sdk.Bool(true),
+		},
+	})
+	require.NoError(t, err)
+
+	externalDatabaseId := sdk.NewExternalObjectIdentifier(secondaryTestClientHelper().Ids.AccountIdentifierWithLocator(), sharedDatabase.ID())
+	err = client.Databases.CreateSecondary(ctx, sharedDatabase.ID(), externalDatabaseId, &sdk.CreateSecondaryDatabaseOptions{})
+	require.NoError(t, err)
+
+	database, err := client.Databases.ShowByID(ctx, sharedDatabase.ID())
+	require.NoError(t, err)
+
+	return database, testClientHelper().Database.DropDatabaseFunc(t, sharedDatabase.ID())
+}
+
+func createApplicationPackage(t *testing.T) (*sdk.ApplicationPackage, func()) {
```

Comment:
why not `CreateApplicationPackage` from the helper client?

Reply:
Actually, this involves a more complex setup with dependencies. Moved to a different file.